packages/cli/generators/controller/templates/src/controllers/controller-rest-template.ts.ejs

```diff
@@ -24,26 +23,26 @@ export class <%= className %>Controller {
   }
 
   @get('<%= httpPathName %>/count')
-  async count(@param.query.string('where') where: Where): Promise<number> {
+  async count(@param.query.string('where') where?: Where): Promise<number> {
     return await this.<%= repositoryNameCamel %>.count(where);
   }
 
   @get('<%= httpPathName %>')
-  async find(@param.query.string('filter') filter: Filter)
+  async find(@param.query.string('filter') filter?: Filter)
     : Promise<<%= modelName %>[]> {
     return await this.<%= repositoryNameCamel %>.find(filter);
   }
 
   @patch('<%= httpPathName %>')
   async updateAll(
-    @param.query.string('where') where: Where,
     @requestBody() obj: <%= modelName %>
+    @param.query.string('where') where?: Where,
   ): Promise<number> {
-    return await this.<%= repositoryNameCamel %>.updateAll(where, obj);
+    return await this.<%= repositoryNameCamel %>.updateAll(obj, where);
```

Comment:
Should the obj to be `Partial<<%= modelName %>>`?

Reply:
Yeah it should be, but from what i remember of #1179, this should incorrectly generate the schema, so it's something that should probably be avoided until it can be fixed

Reply:
I see. It's probably time to introduce a `@type` decorator.